Add a android log implementation

kirklandsign opened the pull request.

larryliu0820 commented on runtime/platform/targets.bzl:
```diff
@@ -41,10 +41,12 @@ def define_common_targets():
-        }),
+        srcs = select({
+            "ovr_config//os:android": ["default/android.cpp"],
+            "DEFAULT": _select_pal({
+                "minimal": ["default/minimal.cpp"],
+                "posix": ["default/posix.cpp"],
```
I'm wondering if this should be another "android" entry

kirklandsign replied:
> However, it might also cause some confusion for users since executorch.pal_default is now ignored

That makes sense, but ovr_config//os:android is automatic.

larryliu0820 commented on runtime/platform/targets.bzl:
```diff
@@ -41,12 +41,15 @@ def define_common_targets():
+        })}),
         deps = [
             ":pal_interface",
+            "fbsource//third-party/toolchains:log",
```
This will break in OSS buck right? Can you also hide it under ovr_config android?

kirklandsign committed:
Add a android log implementation (#10938)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/10938

Differential Revision: D74865527

Pulled By: kirklandsign

jathu reviewed (approved):
build changes lgtm
